cmd/kube-apiserver/app/server.go

```diff
@@ -455,6 +456,7 @@ func (s *APIServer) Run(_ []string) error {
 						glog.Infof("Using self-signed cert (%s, %s)", s.TLSCertFile, s.TLSPrivateKeyFile)
 					}
 				}
+				_ = systemd.SdNotify("READY=1\n")
```

Comment:
It's not just about running under systemd. The service must also have been setup with ```Type=notify```. In addition another init system could implement the same ```NOTIFY_SOCKET``` protocol ... it's intentionally designed to be portable.